src/Integrations/ActiveCampaign/ActiveCampaignClient.php

```diff
@@ -0,0 +1,675 @@
+		$details = $this->getApiReponseDetails($response);
+		$body = $details['body'];
+
+		// Bailout if fields are missing.
+		if (!isset($body['form']['cfields'])) {
+			return [];
+		}
+
+		// Prepeare custom actions.
+		$actions = [];
+
+		if (isset($body['form']['actiondata']['actions'])) {
+			// Map all actions.
+			foreach ($body['form']['actiondata']['actions'] as $item) {
+				$type = $item['type'] ?? '';
+
+				if (!$type) {
+					continue;
+				}
+
+				// Map api qesponse with our naming.
+				switch ($type) {
+					case 'add-a-tag':
+						$actions[] = [
+							'action' => 'tags',
+							'value' => $item['tag'] ?? '',
+						];
+						break;
+					case 'subscribe-to-list':
+						$actions[] = [
+							'action' => 'lists',
+							'value' => $item['list'] ?? '',
+						];
+						break;
+				}
+			}
+		}
+
+		// Bailout with correct data.
+		return [
+			'fields' => $body['form']['cfields'],
+			'actions' => $actions,
+		];
+	}
+
+	/**
+	 * API request to get all lists from ActiveCampaign.
+	 *
+	 * @return array<string, mixed>
+	 */
+	private function getActiveCampaignLists()
+	{
+		$response = \wp_remote_get(
+			"{$this->getBaseUrl()}forms",
+			[
+				'headers' => $this->getHeaders(),
+				'timeout' => 60,
+			]
+		);
+
+		// Output response details.
+		$details = $this->getApiReponseDetails($response);
+		$body = $details['body'];
+
+		if (!isset($body['forms'])) {
+			return [];
+		}
+
+		return $body['forms'];
+	}
+
+	/**
+	 * Prepare params
+	 *
+	 * @param array<string, mixed> $params Params.
+	 *
+	 * @return array<string, mixed>
+	 */
+	private function prepareParams(array $params): array
+	{
+		$output = [];
+
+		$standardFields = \array_flip(ActiveCampaign::STANDARD_FIELDS);
+
+		// Remove unecesery storage field.
+		if (isset($params['es-form-storage'])) {
+			unset($params['es-form-storage']);
+		}
+
+		// Remove unecesery action tags field.
+		if (isset($params['actionTags'])) {
+			unset($params['actionTags']);
+		}
+
+		// Remove unecesery action lists field.
+		if (isset($params['actionLists'])) {
+			unset($params['actionLists']);
+		}
+
+		// Map params.
+		foreach ($params as $key => $param) {
+			$value = $param['value'] ?? '';
+
+			if (!$value) {
+				continue;
+			}
+
+			// If standard key use different logic.
+			if (isset($standardFields[$key])) {
+				// On full name explode first space and output it as first and last name.
+				if ($key === 'fullName') {
+					$value = \explode(' ', $value, 2);
+					$output['firstName'] = $value[0] ?? '';
+					$output['lastName'] = $value[1] ?? '';
+				} else {
+					$output[$key] = $value;
+				}
+			} else {
+				// Mape custom fields.
+				$output['fieldValues'][] = [
+					'field' => $key,
+					'value' => $value,
+				];
+			}
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Return ActiveCampaign base url.
+	 *
+	 * @return string
+	 */
+	private function getBaseUrl(): string
+	{
+		$url = \rtrim($this->getApiUrl(), '/');
+
+		return "{$url}/api/3/";
```

Comment:
Why not this?
```suggestion
		return "{$this->getApiUrl()}api/3/";
```

Reply:
This is used to ensure that the user can provide the API url in setting with and without / at the end

Reply:
Okay, but in some places you do exactly this, meaning some requests could fail if a user enters the API URL with a trailing slash. I think the best course of action would be to sanitize it to a version that suits you best (e.g. without a trailing slash) when saving the API URL.